File: IT/1/Containers/C++/testSet.cpp
/* 
file testSet.cpp
compilare con: g++ -std=c++11 testSet.cpp
*/

#include <iostream>
#include <iterator>
#include <set>

using std::string;
using std::cout;
using std::endl;
using std::ostream_iterator;
using std::set;

int main()
{
	/* dichiara un set di numeri interi: i numeri sono aggiunti in ordine casuale, ma il set li riordina automaticamente secondo l'ordinamento naturale */

	set<int> n = {2, 0, 1};

	cout << "N = [";
	copy(n.begin(), n.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	/* verifica la presenza dello zero nel set */

	if (n.count(0)) {
		cout << "L'insieme N include lo zero." << endl;
	}

	/* aggiunge il numero 3 al set */

	n.insert(3);

	/* scorre tutti gli elementi dell'insieme con un iteratore */

	cout << "N = [";

	for (auto it = n.begin(); it != n.end(); ++it) {
		cout << *it << ", ";
	}

	cout << "\b\b]" << endl;

	/* aggiunge i numeri 4, 5, 6, 7 al set */

	set<int> n2 = {4, 5, 6, 7};

	n.insert(n2.begin(), n2.end());

	cout << "N = [";
	copy(n.begin(), n.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	/* recupera il primo elemento (lo zero) dal set, con un iteratore */

	auto p = n.begin();
	cout << "Primo elemento di N: " << (*p) << endl;

	/* cancella il numero 7 dal set */

	n.erase(7);

	cout << "N = [";
	copy(n.begin(), n.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	/* cancella tutti gli elementi del set */

	n.clear();
	cout << "Cardinalità dell'insieme vuoto: " << n.size() << endl;
}

File: IT/1/Strutture dati/Python/testSet.py
# file testSets.py
# -*- coding: utf-8 -*-

from __future__ import print_function
from sys import stdout
from os import linesep

# dichiara un set di numeri interi: i numeri sono aggiunti in ordine casuale, ma il set li riordina automaticamente secondo l'ordinamento naturale

n = {2, 0, 1}
print(n)

# verifica la presenza dello zero nel set

if 0 in n:
	print("L'insieme N include lo zero.")

# aggiunge il numero 3 al set

n.add(3)

# scorre tutti gli elementi dell'insieme con un ciclo for

for m in n:
	print(m, end=" ")

stdout.write(linesep)

# aggiunge i numeri 4, 5, 6, 7 al set

n.update([5, 4, 7, 6])
print(n)

# recupera il primo elemento (lo zero) dal set

p = n.pop()
print("Primo elemento di N:", p)

# cancella il numero 7 dal set

n.remove(7)
print(n)

# cancella tutti gli elementi del set

n.clear()
print("Cardinalità dell'insieme vuoto:", len(n))
print(n)

